tests/tests.rs

```diff
@@ -500,3 +586,223 @@ fn should_canonicalize_json_response() {
+        .wait().unwrap();
+    assert_eq!(
+        response,
+        Some(candid_types::TransactionReceipt {
+            status: 0.into(),
+            transaction_hash: "0xdd5d4b18923d7aae953c7996d791118102e889bea37b48a651157a4890e4746f"
+                .to_string(),
+            block_number: 18_515_371_u64.into(),
+            block_hash: "0x5115c07eb1f20a9d6410db0916ed3df626cfdab161d3904f45c8c8b65c90d0be"
+                .to_string(),
+            effective_gas_price: 26_776_497_782_u64.into(),
+            gas_used: 32_137.into(),
+        })
+    );
+}
+
+#[test]
+fn eth_get_transaction_count_should_succeed() {
+    let setup = EvmRpcSetup::new().authorize_caller(Auth::FreeRpc);
+    let response = setup
+        .eth_get_transaction_count(
+            CandidRpcSource::EthMainnet(None),
+            candid_types::GetTransactionCountArgs {
+                address: "0xdAC17F958D2ee523a2206206994597C13D831ec7".to_string(),
+                block: candid_types::BlockSpec::Tag(candid_types::BlockTag::Latest),
+            },
+        )
+        .mock_http(MockOutcallBuilder::new(
+            200,
+            r#"{"jsonrpc":"2.0","id":0,"result":"0x1"}"#,
+        ))
+        .wait()
+        .unwrap();
+    assert_eq!(response, 1);
+}
+
+#[test]
+fn eth_fee_history_should_succeed() {
+    let setup = EvmRpcSetup::new().authorize_caller(Auth::FreeRpc);
+    let response = setup
+        .eth_fee_history(
+            CandidRpcSource::EthMainnet(None),
+            candid_types::FeeHistoryArgs {
+                block_count: 3,
+                newest_block: candid_types::BlockSpec::Tag(candid_types::BlockTag::Latest),
+                reward_percentiles: None,
+            },
+        )
+        .mock_http(MockOutcallBuilder::new(
+            200,
+            r#"{"id":0,"jsonrpc":"2.0","result":{"oldestBlock":"0x11e57f5","baseFeePerGas":["0x9cf6c61b9","0x97d853982","0x9ba55a0b0","0x9543bf98d"],"reward":[["0x0123"]]}}"#,
+        ))
+        .wait()
+        .unwrap();
+    assert_eq!(
+        response,
+        Some(FeeHistory {
+            oldest_block: CheckedAmountOf::new(0x11e57f5),
+            base_fee_per_gas: vec!["0x9cf6c61b9", "0x97d853982", "0x9ba55a0b0", "0x9543bf98d"]
+                .into_iter()
+                .map(|hex| CheckedAmountOf::from_str_hex(hex).unwrap())
+                .collect(),
+            reward: vec![vec![CheckedAmountOf::new(0x0123)]],
+        })
+    );
+}
+
+#[test]
+fn eth_send_raw_transaction_should_succeed() {
```

Comment:
Would it make sense to add a few tests for the unhappy path? For example, sending an invalid transaction should result in an error, right?

Reply:
I was planning to add these separately, but I'll include them here for efficiency given the review time zone offset.